index.js

```diff
@@ -3983,6 +3983,51 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Takes a pattern and a string, and returns the result of splitting the
+  //. string at every non-overlapping occurrence of the pattern.
+  //.
+  //. Properties:
+  //.
+  //.   - `forall s :: forall s :: String, t :: String.
+  //.     S.joinWith(s, S.splitOnRegex(S.regex('', S.regexEscape(s)), t)) = t
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/[,;][ ]*/, 'foo, bar, baz')
+  //. ['foo', 'bar', 'baz']
+  //.
+  //. > S.splitOnRegex(/[,;][ ]*/, 'foo;bar;baz')
+  //. ['foo', 'bar', 'baz']
+  //. ```
+  function splitOnRegex(pattern, s) {
+    var $pattern = regex(
+      'g' + (pattern.ignoreCase ? 'i' : '') + (pattern.multiline ? 'm' : ''),
+      pattern.source
+    );
```

Comment:
It occurred to me that it would be semantically accurate for the type of the first argument to be `GlobalRegExp` rather than simply `RegExp`. We are, after all, "splitting […] at *every* non-overlapping occurrence of the pattern".

If we make this change we'll be able to use `withRegex` to avoid cloning `pattern`:

```javascript
return withRegex(pattern, function() {
  var result = [];
  ...
  return result;
});
```